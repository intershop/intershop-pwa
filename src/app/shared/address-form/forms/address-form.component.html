<div [formGroup]="parentForm" data-address-form="data-address-form">
  <p class="indicates-required"><span class="required">*</span>Required field</p>

  <!-- country select -->
  <fieldset>
    <ish-select-country
      [countries]="countries"
      [form]="parentForm"
      controlName="countryCodeSwitch"
      label="Country"
    ></ish-select-country>
  </fieldset>

  <!-- country dependent address forms -->
  <ng-container [ngSwitch]="countryCode">
    <ish-address-form-de *ngSwitchCase="'DE'" [addressForm]="addressForm" [countryCode]="countryCode"></ish-address-form-de>
    <ish-address-form-fr *ngSwitchCase="'FR'" [addressForm]="addressForm" [countryCode]="countryCode"></ish-address-form-fr>
    <ish-address-form-gb *ngSwitchCase="'GB'" [addressForm]="addressForm" [countryCode]="countryCode"></ish-address-form-gb>
    <ish-address-form-us *ngSwitchCase="'US'" [addressForm]="addressForm" [countryCode]="countryCode" [regions]="regions"></ish-address-form-us>
    <ish-address-form-default *ngSwitchDefault [addressForm]="addressForm" [countryCode]="countryCode" [regions]="regions"></ish-address-form-default>
  </ng-container>

  <ish-input
    controlName="phoneHome"
    [form]="addressForm"
    label="Phone"
  ></ish-input>

</div>
