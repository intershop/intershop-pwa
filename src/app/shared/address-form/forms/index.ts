import { ADDRESS_FACTORY } from './address-form.factory';

import { AddressDEComponent } from './address-de/address-de.component';
import { AddressDefaultComponent } from './address-default/address-default.component';
import { AddressFormComponent } from './address-form.component';
import { AddressFRComponent } from './address-fr/address-fr.component';
import { AddressGBComponent } from './address-gb/address-gb.component';
import { AddressUSComponent } from './address-us/address-us.component';

import { AddressDEFactory } from './address-de/address-de.factory';
import { AddressDefaultFactory } from './address-default/address-default.factory';
import { AddressFRFactory } from './address-fr/address-fr.factory';
import { AddressGBFactory } from './address-gb/address-gb.factory';
import { AddressUSFactory } from './address-us/address-us.factory';

export const components = [
  AddressFormComponent,
  AddressDefaultComponent,
  AddressDEComponent,
  AddressFRComponent,
  AddressGBComponent,
  AddressUSComponent,
];

const factories = [
  AddressDefaultFactory,
  AddressDEFactory,
  AddressFRFactory,
  AddressGBFactory,
  AddressUSFactory,
];

/**********************************/

export const factoryProviders = factories.map(f => (
  { provide: ADDRESS_FACTORY, useClass: f, multi: true }
));
