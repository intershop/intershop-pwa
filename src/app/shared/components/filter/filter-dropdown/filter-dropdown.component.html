<div ngbDropdown autoClose="outside">
  <a
    ngbDropdownToggle
    class="form-control"
    role="button"
    id="dropdownMenuLink"
    data-toggle="dropdown"
    aria-haspopup="true"
    aria-expanded="false"
    [attr.title]="placeholder !== filterElement.name ? placeholder : null"
  >
    <span>{{ placeholder }}</span>
  </a>
  <div ngbDropdownMenu aria-labelledby="dropdownMenuLink">
    <a
      *ngFor="let facet of filterElement.facets; trackBy: trackByFn"
      class="dropdown-item"
      [ngClass]="{ selected: facet.selected }"
      (click)="apply(facet)"
    >
      {{ facet.displayName }}
      <ng-container *ngIf="facet.count"> ({{ facet.count }})</ng-container>
      <fa-icon *ngIf="facet.selected" [icon]="['fas', 'check']" class="icon-checked"></fa-icon>
    </a>
  </div>
</div>
