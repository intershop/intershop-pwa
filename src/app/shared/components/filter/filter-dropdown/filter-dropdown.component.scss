a {
  display: flex;
  padding-right: 20px;

  span:first-child {
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
}
