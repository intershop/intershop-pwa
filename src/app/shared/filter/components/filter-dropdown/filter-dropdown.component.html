<ul class="filter-list">
  <ng-container *ngFor="let facet of filterElement.facets">
    <li *ngIf="facet.selected" class="filter-item filter-selected">
      <span class="icon-selected"></span>{{ facet.displayName }} <span class="filter-count">({{ facet.count }})</span>
      <a class="filter-clear" (click)="filter(facet.name)">
        <fa-icon class="float-right" [icon]="['fas', 'minus']"></fa-icon>
      </a>
    </li>
  </ng-container>
</ul>

<div [attr.data-testing-id]="'collapse-filter-' + (filterElement.name | sanitize)">
  <form role="filter" name="Filter" [formGroup]="filterForm">
    <select formControlName="filterDropdown" name="SearchParameter" class="ws-submit-search-param form-control">
      <option value="">{{ 'search.filter.select_option.text' | translate }}</option>
      <ng-container *ngFor="let facet of filterElement.facets">
        <option *ngIf="!facet.selected" [value]="facet.name">{{ facet.displayName }}</option>
      </ng-container>
    </select>
  </form>
</div>
