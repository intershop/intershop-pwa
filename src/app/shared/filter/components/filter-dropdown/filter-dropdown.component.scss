a {
  display: flex;
  padding-right: 20px;

  span:first-child {
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  span:last-child {
    padding-left: 5px;
  }

  &.selected {
    background-color: rgb(96, 251, 251);
  }
}
