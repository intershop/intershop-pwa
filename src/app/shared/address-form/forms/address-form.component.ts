import { Component, Input, OnInit } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { Country } from '../../../models/country/country.model';
import { Region } from '../../../models/region/region.model';

@Component({
  selector: 'ish-address-form',
  templateUrl: './address-form.component.html'
})
export class AddressFormComponent implements OnInit {

  @Input() parentForm: FormGroup;
  @Input() controlName = 'address';
  @Input() countryCode: any;
  @Input() countries: Country[];
  @Input() regions: Region[];

  ngOnInit() {
    if (!this.parentForm) {
      throw new Error('required input parameter <parentForm> is missing for AddressFormComponent');
    }
  }

  get addressForm(): FormGroup {
    return this.parentForm.get(this.controlName) as FormGroup;
  }
}
