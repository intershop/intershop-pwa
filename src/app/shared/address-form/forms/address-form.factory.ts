import { InjectionToken } from '@angular/core';
import { FormGroup } from '@angular/forms';

export const ADDRESS_FORM_FACTORY = new InjectionToken<AddressFormFactory>('Address Form Factory');

export abstract class AddressFormFactory {
  countryCode = 'default';
  countryLabel = '';

  group(): FormGroup {
    return new FormGroup({});
  }

  getGroup(value?: { [key: string]: any }): FormGroup {
    const newGroup = this.group();
    if (value) { newGroup.patchValue(value); }
    return newGroup;
  }

}
