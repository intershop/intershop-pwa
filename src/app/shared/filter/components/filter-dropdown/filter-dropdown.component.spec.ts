import { ComponentFixture, TestBed, async, fakeAsync, tick } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { FaIconComponent } from '@fortawesome/angular-fontawesome';
import { NgbCollapse } from '@ng-bootstrap/ng-bootstrap';
import { TranslateModule } from '@ngx-translate/core';
import { MockComponent } from 'ng-mocks';

import { Filter } from 'ish-core/models/filter/filter.model';
import { PipesModule } from 'ish-core/pipes.module';

import { FilterDropdownComponent } from './filter-dropdown.component';

describe('Filter Dropdown Component', () => {
  let component: FilterDropdownComponent;
  let fixture: ComponentFixture<FilterDropdownComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [PipesModule, ReactiveFormsModule, TranslateModule.forRoot()],
      declarations: [FilterDropdownComponent, MockComponent(FaIconComponent), MockComponent(NgbCollapse)],
    }).compileComponents();
  }));

  beforeEach(() => {
    const filterElement = {
      name: 'Brands',
      facets: [
        { name: 'AsusName', count: 4, displayName: 'Asus' },
        { name: 'LogitechName', count: 5, displayName: 'Logitech', selected: true },
      ],
    } as Filter;

    fixture = TestBed.createComponent(FilterDropdownComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    component.filterElement = filterElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });

  it('should toggle unselected filter facets when filter group header is clicked', fakeAsync(() => {
    fixture.detectChanges();
    const filterGroupHead = fixture.nativeElement.querySelectorAll('h3')[0];
    filterGroupHead.click();
    tick(500);
    fixture.detectChanges();

    const selectedFilterFacet = element.getElementsByClassName('filter-selected')[0];
    expect(selectedFilterFacet.textContent).toContain('Logitech');

    const hiddenFilters = element.querySelector('[data-testing-id=collapse-filter-Brands]');
    expect(hiddenFilters.className).not.toContain('show');
  }));
});
