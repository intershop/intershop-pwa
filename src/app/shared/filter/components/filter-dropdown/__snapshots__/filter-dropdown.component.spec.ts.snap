// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Dropdown Component should be created 1`] = `
<div class="filter-group">
  <h3 aria-controls="collapseFilter" aria-expanded="true">
    Brands <fa-icon class="float-right" ng-reflect-icon-prop="fas,angle-down"></fa-icon>
  </h3>
  <ul class="filter-list">
    <li class="filter-item filter-selected">
      <span class="icon-selected"></span>Logitech <span class="filter-count">(5)</span
      ><a class="filter-clear"
        ><fa-icon class="float-right" ng-reflect-icon-prop="fas,minus"></fa-icon
      ></a>
    </li>
  </ul>
  <div id="collapseFilter" ng-reflect-collapsed="false" data-testing-id="collapse-filter-Brands">
    <form name="Filter" novalidate="" role="filter" class="ng-untouched ng-pristine ng-valid">
      <select
        class="ws-submit-search-param form-control ng-untouched ng-pristine ng-valid"
        formcontrolname="filterDropdown"
        name="SearchParameter"
        ng-reflect-name="filterDropdown"
        ><option value="" ng-reflect-value="">search.filter.select_option.text</option
        ><option value="AsusName" ng-reflect-value="AsusName">Asus</option></select
      >
    </form>
  </div>
</div>
`;
